Sources/MapboxMaps/Foundation/ResourceOptionsManager.swift

```diff
@@ -0,0 +1,151 @@
+import Foundation
+@_implementationOnly import MapboxCommon_Private
+
+/// Convenience class that manages a global `ResourceOptions`
+///
+/// It's possible to create `ResourceOptionsManager` instances as you need them,
+/// however it's convenient to use the default object (`default`).
+///
+/// For example, we recommend that the Mapbox access token be set in
+/// `application(_:didFinishLaunchingWithOptions:)` rather than relying on the
+/// value in your application's Info.plist:
+///
+///     ```
+///     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+///         // Override point for customization after application launch.
+///         ResourceOptionsManager.default.update { resourceOptions in
+///             resourceOptions.accessToken = "overridden-access-token"
+///         }
+///         return true
+///     }
+///     ```
+public class ResourceOptionsManager {
+
+    /// Default instance
+    ///
+    /// This shared instance is used by the default initializers
+    /// for `ResourceOptions` and `MapInitOptions`.
+    ///
+    /// The application's Info.plist will be searched for a valid access token
+    /// under the key `MBXAccessToken`
+    ///
+    /// A valid access token must be provided or found.
+    public static var `default`: ResourceOptionsManager {
+        if defaultInstance == nil {
+            defaultInstance = ResourceOptionsManager(accessToken: nil)
+        }
+        return defaultInstance!
+    }
+
+    /// :nodoc:
+    /// Convenience function to remove the default instance. Calling `default`
+    /// again will re-create the default instance.
+    internal static func destroyDefault() {
+        defaultInstance = nil
+    }
```

Comment:
This is in preparation of being able to clean up TileStores completely - useful for tests.

Reply:
we are considering to expose the same cc @Chaoba